### ISSUE TITLE
Problems with Game.Build.cs (regarding C++)

### ISSUE THREAD

**Issue by Ryder17z** [FlaxEngine/FlaxEngine#2139]
**Issue description:**
It's pretty confusing why the "C++ script" template appears broken out-of-the-box.

And this is the culprit:
![image](https://github.com/FlaxEngine/FlaxEngine/assets/2000703/28696481-29fb-4a78-88be-b0d80bab2421)

I read that comment as when true, no C++ support and only C# which is backwards compared to what setting the value to true does.

**Steps to reproduce:**
create a new flax project, then in the editor, right-click in the content browser to create a new C++ script and open that in the editor.

**Minimal reproduction project:**
N/A (comes with engine)

**Flax version:**
1.7.6

**Proposal**
Change the comment to better reflect behavior. i.e.  "Enable C++ support"

Also; what would be the implications of setting the value to true by default?

**Comment by MineBill** [comment]
Ideally the name of that property should be changed to something like `BuildCppCode` or something similar but that would be a breaking change.

I propose a change to the comment in the template:
```cs
// Change to this to true to enable
// compilation for C++ scripts.
BuildNativeCode = false;
```

**Comment by Ryder17z** [comment]
Come to think of it, to avoid breaking things, one could do i.e.

`#define BuidNativeCode BuildCppCode `

**Comment by mafiesto4** [comment]
Yeah Editor could check if C++ scripting was enabled for a game and prevent creating incorrect scripts or just auto-enable it manually in build files for simplicity.
